src/routes/stats/idea.js

```diff
@@ -1,18 +1,30 @@
 const config    = require('config');
 const dbConfig  = config.get('database');
-const mysql = require('mysql2');
+const mysql = require('mysql2/promise');
 const express = require('express');
 const createError = require('http-errors')
-
-const pool = mysql.createPool({
-  host: dbConfig.host,
-  user: dbConfig.user,
-  password: dbConfig.password,
-  database: dbConfig.database,
-  waitForConnections: true,
-  connectionLimit: 10,
-  queueLimit: 0
-});
+const getAzureAuthToken = require('../../util/azure-auth')
+
+const createPool = async () => {
```

Comment:
Are we setting up a new db connection pool on every get request here? What I think you want is to setup a connection pool once and reuse it for all requests.

We can extract the createPool functions to its own module. We can also put the mysql createconnections there. Although we can probably utilize the pool for that connection too. The pool is just a bundle of db connections right?

On application startup, we can setup the db connection pool once. Then await that connection before starting up the rest of the application (and allowing requests). After the await you are guaranteed to have the pool and you don't have to await it anymore